### ISSUE TITLE
<ion-icon name="?"> elements won't be displayed in AlertController

### ISSUE THREAD

**Issue by redwolf2** [ionic-team/ionic#18239]
**Ionic version:**
[x] **4.x**

**I'm submitting a ...** 
[x] bug report
[ ] feature request

**Current behavior:**

`ion-icon` elements displayed in the AlertController won't show the icon, if the `name` attribute is used. SVG files with the `src` attributes work.

![grafik](https://user-images.githubusercontent.com/4141621/57468295-783ca280-7284-11e9-9133-1907175fdebb.png)

I am sure I saw this working in an older version of Ionic 4.

**Related code:**

[ion-icon-bug.zip](https://github.com/ionic-team/ionic/files/3163055/ion-icon-bug.zip)

```typescript
import { AlertController } from '@ionic/angular';

public constructor(public alertController: AlertController) {

}

public async showInfoDialog() {
  const prompt = await this.alertController.create({
    header: 'name does not work, src does',
    message:
    '<div><ion-icon name="body"></ion-icon></div>' +
    '<div><ion-icon src="assets/icon/truck.svg"></ion-icon></div>',
     buttons: [{ text: 'OK' }]
   });
   await prompt.present();
}
```

**Ionic info:** 

```
Ionic:

   ionic (Ionic CLI)             : 4.12.0 (C:\Users\user\AppData\Roaming\npm\node_modules\ionic)
   Ionic Framework               : @ionic/angular 4.4.0
   @angular-devkit/build-angular : 0.13.9
   @angular-devkit/schematics    : 7.3.9
   @angular/cli                  : 7.3.9
   @ionic/angular-toolkit        : 1.5.1

System:

   NodeJS : v10.15.3 (C:\Program Files\nodejs\node.exe)
   npm    : 6.4.1
   OS     : Windows 7
```

**Comment by brandyscarney** [comment]
Thanks for the issue! I was able to confirm this: https://codepen.io/brandyscarney/pen/qGZydB?editors=1010

The following error is being thrown: `icon was not resolved`

**Comment by topalavlad** [comment]
This happens because the message is sanitized:
`innerHTML={sanitizeDOMString(this.message)}` which only allows the following attributes: `const allowedAttributes = ['class', 'id', 'href', 'src'];` and removes the others.

Another example is badge's slot attribute: https://codepen.io/Dudelozaur/pen/QREwJw
You can see that the badges are placed differently on the alert than the initial page. Same thing happens for button shapes, setting items as disabled.

**Comment by redwolf2** [comment]
@topalavlad Thanks, I understand the reason. Do you think there should be a way to pass unsansitized data to the alert dialog? Maybe by using an alternative type like [SafeHtml](https://angular.io/api/platform-browser/SafeHtml) that could be passed as the message?

**Comment by liamdebeasi** [comment]
Hi everyone,

This issue has been resolved via #18246 and will be in the next version of Ionic. I added in the `name` and `slot` attributes to the allowed list of attributes since I felt those were commonly used attributes.

In general, we don't add attributes to the allowed list of attributes. Components that allow HTML content like `ion-alert` are not intended for anything other than some basic HTML.

I have created a feature request for giving users the option of bypassing the sanitizer if they need to do something complex: https://github.com/ionic-team/ionic/issues/18277

Thanks!
